src/software/SfM/main_ComputeFeatures.cpp

```diff
@@ -213,11 +231,14 @@ int main(int argc, char **argv)
     Image<unsigned char> imageGray;
     C_Progress_display my_progress_bar( sfm_data.GetViews().size(),
       std::cout, "\n- EXTRACT FEATURES -\n" );
-    for(Views::const_iterator iterViews = sfm_data.views.begin();
-        iterViews != sfm_data.views.end();
-        ++iterViews, ++my_progress_bar)
+
+#ifdef OPENMVG_USE_OPENMP
+    omp_set_num_threads(iNumThreads);
+    #pragma omp parallel for schedule(dynamic) firstprivate(imageGray) if(iNumThreads > 0)
+#endif
+    for(int i = 0; i < sfm_data.views.size(); ++i)
     {
-      const View * view = iterViews->second.get();
+      const View * view = sfm_data.views[i].get();
```

Comment:
In case of none contiguous index in sfm_data it's better to use something like:
```
Views::const_iterator iterViews = sfm_data.views.begin();
std::advance(iterViews, i);
const View * view = iterViews->second.get();
```